cli/src/cmd_auth.rs

```diff
@@ -353,46 +376,42 @@ pub struct CmdAuthLogout {
 }
 
 impl CmdAuthLogout {
-    pub async fn run(&self, ctx: &Context) -> Result<()> {
-        if !self.force {
-            match dialoguer::Confirm::new()
-                .with_prompt("Confirm authentication information deletion:")
-                .interact()
-            {
-                Ok(true) => {}
-                Ok(false) => {
-                    println!("Action aborted. No changes have been made.");
-                    return Ok(());
-                }
-                Err(err) => {
-                    return Err(anyhow!("prompt failed: {}", err));
-                }
-            }
+    pub async fn logout(&self, ctx: &Context) -> Result<()> {
+        if !self.force && !yes("Confirm authentication information deletion:")? {
+            return Ok(());
         }
 
-        match &self.host {
-            Some(host) => {
-                // Setting the host with empty parameters so it will now be listed as
-                // "unauthenticated" when running `$ oxide auth status`.
-                let host_entry = Host {
-                    token: String::from(""),
-                    user: String::from(""),
-                    default: false,
-                };
-                ctx.config().update_host(host.to_string(), host_entry)?;
+        let credentials_path = ctx.client_config().config_dir().join("credentials.toml");
 
-                println!("Removed authentication information for: {}", host);
-            }
-            None => {
-                let mut dir = dirs::home_dir().unwrap();
-                dir.push(".config");
-                dir.push("oxide");
-                let hosts_path = dir.join("hosts.toml");
-
-                // Clear the entire file for users who want to reset their known hosts.
-                let _ = File::create(hosts_path)?;
-                println!("Removed all authentication information");
+        if self.all {
+            // Clear the entire file for users who want to reset their known hosts.
+            let _ = File::create(credentials_path)?;
+            println!("Removed all authentication information");
+        } else {
+            let profile = ctx
+                .client_config()
+                .profile()
+                .or_else(|| ctx.config_file().basics.default_profile.as_deref());
+            let Some(profile_name) = profile else {
+                bail!("No profile specified and no default profile");
+            };
```

Comment:
Does this mean that if no profile name is selected, it'll just remove information of the default profile? Maybe we could add a warning or something? Or disallow logging out without setting the profile to be logged out of?

Reply:
Does this PR change things in that regard?